core/src/test/java/org/mule/execution/ErrorHandlingExecutionTemplateTestCase.java

```diff
@@ -271,6 +276,17 @@ public void testTransactionIsResolved() throws Exception
         assertThat(TransactionCoordination.getInstance().getTransaction(), IsNull.<Object>nullValue());
     }
 
+    @Test
+    public void transactionCommitFailsCallsExceptionHandler() throws Exception
+    {
+        ExecutionTemplate executionTemplate = createExceptionHandlingTransactionTemplate();
+        TransactionCoordination.getInstance().bindTransaction(mockTransaction);
+        configureExceptionListener(NO_ROLLBACK_FILTER, NO_COMMIT_FILTER);
+        executionTemplate.execute(TransactionTemplateTestUtils.getEmptyTransactionCallback(mockEvent));
+        verify(mockTransaction, never()).commit();
+        verify(mockTransaction, never()).rollback();
```

Comment:
Looks like there is a missing assertion checking that the exceptionHandler was invoked, otherwise the test name is not clear